client-react/src/pages/app/change-app-plan/CreateOrSelectPlan.tsx

```diff
@@ -58,12 +61,21 @@ export const CreateOrSelectPlan = (props: CreateOrSelectPlanFormValues & CreateO
     existingPlan,
   });
 
-  const onChangeDropdown = (e: unknown, option: IDropdownOption) => {
+  const onChangeDropdown = (_e: unknown, option: IDropdownOption) => {
     const info = { ...planInfo };
 
     if (option.data === NEW_PLAN) {
       info.isNewPlan = true;
       info.newPlanInfo.name = option.key as string;
+      if (info.newPlanInfo.tier !== skuTier) {
+        if (skuTier && skuTier === ChangeAppPlanTierTypes.Dynamic) {
```

Comment:
The equality check ensures that `skuTier` is truthy.
```suggestion
        if (skuTier === ChangeAppPlanTierTypes.Dynamic) {
```